### PR TITLE
React to Razor.Design package removal

### PR DESCRIPTION
**Pull request by pranavkm**
* Remove references to packages (Razor.Design, Razor.Extensions) solely used to bring in compiler \ targets
* Target netcoreapp3.0 in samples and tests to allow Sdk to infer values

### CODE REVIEW

File: src/Microsoft.Extensions.ApiDescription.Design/Microsoft.Extensions.ApiDescription.Design.csproj [hunk 1]
```diff
@@ -42,6 +42,11 @@
       <PackagePath>tools/dotnet-getdocument.dll</PackagePath>
       <StrongName>$(AssemblySigningStrongName)</StrongName>
     </SignedPackageFile>
+    <SignedPackageFile Include="../dotnet-getdocument/bin/$(Configuration)/netcoreapp2.1/publish/dotnet-getdocument.exe">
```
Comment (pranavkm) [inline src/Microsoft.Extensions.ApiDescription.Design/Microsoft.Extensions.ApiDescription.Design.csproj]: @dougbu the build produces an extra exe in the publish dir. Filing https://github.com/aspnet/Mvc/issues/8681 to investigate this later.

File: build/dependencies.props [hunk 2]
```diff
@@ -16,88 +16,88 @@
...
+    <MicrosoftExtensionsValueStopwatchSourcesPackageVersion>3.0.0-alpha1-10657</MicrosoftExtensionsValueStopwatchSourcesPackageVersion>
+    <MicrosoftExtensionsWebEncodersPackageVersion>3.0.0-alpha1-10670</MicrosoftExtensionsWebEncodersPackageVersion>
     <MicrosoftNETCoreApp20PackageVersion>2.0.9</MicrosoftNETCoreApp20PackageVersion>
     <MicrosoftNETCoreApp21PackageVersion>2.1.3</MicrosoftNETCoreApp21PackageVersion>
-    <MicrosoftNETCoreApp22PackageVersion>2.2.0-preview3-27014-02</MicrosoftNETCoreApp22PackageVersion>
-    <MicrosoftNetHttpHeadersPackageVersion>3.0.0-alpha1-10654</MicrosoftNetHttpHeadersPackageVersion>
-    <MicrosoftNETSdkRazorPackageVersion>3.0.0-alpha1-10654</MicrosoftNETSdkRazorPackageVersion>
+    <MicrosoftNETCoreApp30PackageVersion>3.0.0-preview1-26907-05</MicrosoftNETCoreApp30PackageVersion>
+    <MicrosoftNetHttpHeadersPackageVersion>3.0.0-alpha1-10670</MicrosoftNetHttpHeadersPackageVersion>
+    <MicrosoftNETSdkRazorPackageVersion>3.0.0-a-alpha1-sdk-31823</MicrosoftNETSdkRazorPackageVersion>
```
Comment (pranavkm) [inline build/dependencies.props]: Feature branch of Razor Sdk

**Comment by dougbu** [comment]
@pranavkm it's possible .NET Core changed their default formats between 2.2 and 3.0. Since the failure doesn't occur without your changes, I suspect that's the case. The line to change is https://github.com/aspnet/Mvc/blob/c74a945dda2c96af30bb9dd1482a67edfe6fdd56/test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/RouteValueProviderTests.cs#L78

**Comment by dougbu** [comment]
@pranavkm I found the relevant CoreFx change: See issue dotnet/corefx#29538, PR dotnet/coreclr#18316 or commit dotnet/coreclr@a51328304a. Their change is in 3.0 only.

File: benchmarkapps/BasicApi/BasicApi.csproj [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp2.2</TargetFrameworks>
+    <TargetFrameworks>netcoreapp3.0</TargetFrameworks>
```
Comment (dougbu) [inline benchmarkapps/BasicApi/BasicApi.csproj]: What fails if we use `$(StandardTestWebsiteTfms)` in these projects with hard-coded target frameworks? If we can, would be great to get rid of the special cases.

File: test/WebSites/RazorPagesClassLibrary/RazorPagesClassLibrary.csproj [hunk 4]
```diff
@@ -4,6 +4,10 @@
     <TargetFrameworks>netstandard2.0</TargetFrameworks>
     <!--  Workaround https://github.com/dotnet/core-setup/issues/3726 -->
     <GenerateDependencyFile>false</GenerateDependencyFile>
+
+    <_EnableAllInclusiveRazorSdk>true</_EnableAllInclusiveRazorSdk>
+    <RazorLangVersion>2.1</RazorLangVersion>
+    <RazorDefaultConfiguration>MVC-2.1</RazorDefaultConfiguration>
```
Comment (dougbu) [inline test/WebSites/RazorPagesClassLibrary/RazorPagesClassLibrary.csproj]: Why target only 2.1 in this project?

File: src/Microsoft.Extensions.ApiDescription.Design/Microsoft.Extensions.ApiDescription.Design.csproj (hunk 1 above)
Comment (dougbu) [reply src/Microsoft.Extensions.ApiDescription.Design/Microsoft.Extensions.ApiDescription.Design.csproj]: Sounds good 😺

File: build/dependencies.props [hunk 5]
```diff
@@ -16,88 +16,88 @@
...
+    <MicrosoftExtensionsPrimitivesPackageVersion>3.0.0-alpha1-10657</MicrosoftExtensionsPrimitivesPackageVersion>
+    <MicrosoftExtensionsPropertyActivatorSourcesPackageVersion>3.0.0-alpha1-10657</MicrosoftExtensionsPropertyActivatorSourcesPackageVersion>
+    <MicrosoftExtensionsPropertyHelperSourcesPackageVersion>3.0.0-alpha1-10657</MicrosoftExtensionsPropertyHelperSourcesPackageVersion>
+    <MicrosoftExtensionsSecurityHelperSourcesPackageVersion>3.0.0-alpha1-10657</MicrosoftExtensionsSecurityHelperSourcesPackageVersion>
+    <MicrosoftExtensionsTypeNameHelperSourcesPackageVersion>3.0.0-alpha1-10657</MicrosoftExtensionsTypeNameHelperSourcesPackageVersion>
+    <MicrosoftExtensionsValueStopwatchSourcesPackageVersion>3.0.0-alpha1-10657</MicrosoftExtensionsValueStopwatchSourcesPackageVersion>
+    <MicrosoftExtensionsWebEncodersPackageVersion>3.0.0-alpha1-10670</MicrosoftExtensionsWebEncodersPackageVersion>
     <MicrosoftNETCoreApp20PackageVersion>2.0.9</MicrosoftNETCoreApp20PackageVersion>
     <MicrosoftNETCoreApp21PackageVersion>2.1.3</MicrosoftNETCoreApp21PackageVersion>
-    <MicrosoftNETCoreApp22PackageVersion>2.2.0-preview3-27014-02</MicrosoftNETCoreApp22PackageVersion>
```
Comment (dougbu) [inline build/dependencies.props]: We don't want _any_ tests using 2.2 anymore? Seems extreme and leaves us with a testing gap.

File: build/dependencies.props [hunk 6]
```diff
@@ -16,88 +16,88 @@
...
-    <MicrosoftAspNetCoreAuthenticationCookiesPackageVersion>3.0.0-alpha1-10654</MicrosoftAspNetCoreAuthenticationCookiesPackageVersion>
-    <MicrosoftAspNetCoreAuthenticationCorePackageVersion>3.0.0-alpha1-10654</MicrosoftAspNetCoreAuthenticationCorePackageVersion>
-    <MicrosoftAspNetCoreAuthenticationJwtBearerPackageVersion>3.0.0-alpha1-10654</MicrosoftAspNetCoreAuthenticationJwtBearerPackageVersion>
-    <MicrosoftAspNetCoreAuthenticationPackageVersion>3.0.0-alpha1-10654</MicrosoftAspNetCoreAuthenticationPackageVersion>
-    <MicrosoftAspNetCoreAuthorizationPolicyPackageVersion>3.0.0-alpha1-10654</MicrosoftAspNetCoreAuthorizationPolicyPackageVersion>
-    <MicrosoftAspNetCoreBenchmarkRunnerSourcesPackageVersion>3.0.0-alpha1-10654</MicrosoftAspNetCoreBenchmarkRunnerSourcesPackageVersion>
-    <MicrosoftAspNetCoreChunkingCookieManagerSourcesPackageVersion>3.0.0-alpha1-10654</MicrosoftAspNetCoreChunkingCookieManagerSourcesPackageVersion>
-    <MicrosoftAspNetCoreCookiePolicyPackageVersion>3.0.0-alpha1-10654</MicrosoftAspNetCoreCookiePolicyPackageVersion>
-    <MicrosoftAspNetCoreCorsPackageVersion>3.0.0-a-alpha1-w-m-16569</MicrosoftAspNetCoreCorsPackageVersion>
```
Comment (dougbu) [inline build/dependencies.props]: To confirm: We no longer need to use the feature branch packages for CORS or Routing?

File: benchmarkapps/BasicApi/BasicApi.csproj (hunk 3 above)
Comment (pranavkm) [reply benchmarkapps/BasicApi/BasicApi.csproj]: The variable isn't defined in the root Directory.Build.props (it's in the test Directory.Build.props). So wouldn't be available here. In addition, I think these projects are meant to be buildable standalone on the benchmark server, so I'm going to leave it be for now

File: build/dependencies.props (hunk 6 above)
Comment (pranavkm) [reply build/dependencies.props]: Yeah. Haven't seen any test failures as a result of updating

File: build/dependencies.props (hunk 5 above)
Comment (pranavkm) [reply build/dependencies.props]: I don't think we've ever tested an earlier version of netcoreapp TFM in our repo.

File: test/WebSites/RazorPagesClassLibrary/RazorPagesClassLibrary.csproj (hunk 4 above)
Comment (pranavkm) [reply test/WebSites/RazorPagesClassLibrary/RazorPagesClassLibrary.csproj]: It's the most recent version of Razor that's available

File: test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/RouteValueProviderTests.cs [hunk 7]
```diff
@@ -74,8 +74,7 @@ public void GetValueProvider_ReturnsValue_UsesSpecifiedCulture()
             });
             var provider = new RouteValueProvider(BindingSource.Query, values, new CultureInfo("de-CH"));
 
-            // de-CH culture is slightly different on Windows versus other platforms.
-            var expected = TestPlatformHelper.IsWindows ? "31.10.2018 07:37:38 -07:00" : "31.10.18 07:37:38 -07:00";
+            var expected = "31.10.2018 07:37:38 -07:00";
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/RouteValueProviderTests.cs]: cc @dougbu

Comment (dougbu) [reply test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/RouteValueProviderTests.cs]: 👍

Might be worth a comment for future mergers, alerting developers that the `master` and `release/2.2` branches are _intentionally_ different here.

File: benchmarkapps/BasicApi/BasicApi.csproj (hunk 3 above)
Comment (dougbu) [reply benchmarkapps/BasicApi/BasicApi.csproj]: Ah, now I remember doing that&hellip;

**Pull request merged**